include/ua_plugin_log.h

```diff
@@ -118,6 +121,90 @@ UA_LOG_FATAL(UA_Logger logger, UA_LogCategory category, const char *msg, ...) {
 #define UA_PRINTF_STRING_FORMAT "\"%.*s\""
 #define UA_PRINTF_STRING_DATA(STRING) (int)(STRING).length, (STRING).data
 
+//TODO remove when we merge architectures pull request
+#ifndef UA_snprintf
+# if defined(_WIN32)
+#  define UA_snprintf(source, size, string, ...) _snprintf_s(source, size, _TRUNCATE, string, __VA_ARGS__)
+# else
+#  include <stdio.h>
+#  define UA_snprintf snprintf
+# endif
+#endif
+
+static UA_INLINE UA_StatusCode
+UA_ByteString_toString(const UA_ByteString *byteString, UA_String *str) {
+    if (str->length != 0) {
+        UA_free(str->data);
+        str->length = 0;
+    }
+    if (byteString == NULL)
+        return UA_STATUSCODE_GOOD;
+    //
+    str->length = byteString->length*2;
+    str->data = (UA_Byte*)UA_malloc(str->length+1);
+    if (str->data == NULL)
+        return UA_STATUSCODE_BADOUTOFMEMORY;
+
+    for (size_t i=0; i<byteString->length; i++)
+        UA_snprintf((char*)&str->data[i*2], 2+1, "%02x", byteString->data[i]);
+    return UA_STATUSCODE_GOOD;
+}
+
+static UA_INLINE UA_StatusCode
+UA_NodeId_toString(const UA_NodeId *nodeId, UA_String *nodeIdStr) {
+    if (nodeIdStr->length != 0) {
+        UA_free(nodeIdStr->data);
+        nodeIdStr->length = 0;
+    }
+    if (nodeId == NULL)
+        return UA_STATUSCODE_GOOD;
+
+
+    UA_ByteString byteStr = UA_BYTESTRING_NULL;
+    switch (nodeId->identifierType) {
+        /* for all the lengths below we add the constant for: */
+        /* strlen("ns=XXXXXX;i=")=11 */
+        case UA_NODEIDTYPE_NUMERIC:
+            /* ns (2 byte, 65535) = 5 chars, numeric (4 byte, 4294967295) = 10 chars, delim = 1 , nullbyte = 1-> 17 chars */
+            nodeIdStr->length = 11 + 10 + 1;
+            nodeIdStr->data = (UA_Byte*)UA_malloc(nodeIdStr->length);
+            if (nodeIdStr->data == NULL)
+                return UA_STATUSCODE_BADOUTOFMEMORY;
+            UA_snprintf((char*)nodeIdStr->data, nodeIdStr->length, "ns=%d;i=%lu",
+                        nodeId->namespaceIndex, (unsigned long )nodeId->identifier.numeric);
+            break;
+        case UA_NODEIDTYPE_STRING:
+            /* ns (16bit) = 5 chars, strlen + nullbyte */
+            nodeIdStr->length = 11 + nodeId->identifier.string.length + 1;
+            nodeIdStr->data = (UA_Byte*)UA_malloc(nodeIdStr->length);
+            if (nodeIdStr->data == NULL)
+                return UA_STATUSCODE_BADOUTOFMEMORY;
+            UA_snprintf((char*)nodeIdStr->data, nodeIdStr->length, "ns=%d;i=%.*s", nodeId->namespaceIndex,
```

Comment:
The identifier type is specified as i= for numeric, s= for string, g= for GUID and b= for bytestring  (spec, part 6 page 22).